engine/packages/pegboard/src/workflows/runner2.rs

```diff
@@ -0,0 +1,1176 @@
+								bail!(
+									"received message that should not be sent to runner workflow: {:?}",
+									sig.inner
+								)
+							}
+						}
+					}
+					Main::Command(command) => {
+						// If draining, ignore start actor command and inform the actor wf that it is lost
+						if let (
+							protocol::Command::CommandStartActor(protocol::CommandStartActor {
+								actor_id,
+								generation,
+								..
+							}),
+							true,
+						) = (&command.inner, state.draining)
+						{
+							tracing::warn!(
+								?actor_id,
+								"attempt to schedule actor to draining runner, reallocating"
+							);
+
+							let res = ctx
+								.signal(crate::workflows::actor::Lost {
+									generation: *generation,
+									// Because this is a race condition, we want the actor to reschedule
+									// regardless of its crash policy
+									force_reschedule: true,
+									reset_rescheduling: true,
+								})
+								.to_workflow::<crate::workflows::actor::Workflow>()
+								.tag("actor_id", actor_id)
+								.graceful_not_found()
+								.send()
+								.await?;
+							if res.is_none() {
+								tracing::warn!(
+									?actor_id,
+									"actor workflow not found, likely already stopped"
+								);
+							}
+						} else {
+							commands.push(command.inner);
+						}
+					}
+					Main::CheckQueue(_) => {
+						check_queue = true;
+					}
+					Main::Stop(sig) => {
+						handle_stopping(ctx, &input, state, sig.reset_actor_rescheduling).await?;
+					}
+				}
+			}
+
+			let mut messages = Vec::new();
+
+			// Process init packet
+			if let Some(protocol::ToServerInit {
+				last_command_idx,
+				prepopulate_actor_names,
+				metadata,
+				..
+			}) = init
+			{
+				let init_data = ctx
+					.activity(ProcessInitInput {
+						runner_id: input.runner_id,
+						namespace_id: input.namespace_id,
+						last_command_idx: last_command_idx.unwrap_or(-1),
+						prepopulate_actor_names,
+						metadata,
+					})
+					.await?;
+
+				messages.push(protocol::ToClient::ToClientInit(protocol::ToClientInit {
+					runner_id: input.runner_id.to_string(),
+					last_event_idx: init_data.last_event_idx,
+					metadata: protocol::ProtocolMetadata {
+						runner_lost_threshold: runner_lost_threshold,
+					},
+				}));
+
+				// Send missed commands
+				if !init_data.missed_commands.is_empty() {
+					messages.push(protocol::ToClient::ToClientCommands(
+						init_data.missed_commands,
+					));
+				}
+
+				if !state.draining {
+					ctx.activity(InsertDbInput {
+						runner_id: input.runner_id,
+						namespace_id: input.namespace_id,
+						name: input.name.clone(),
+						key: input.key.clone(),
+						version: input.version,
+						total_slots: input.total_slots,
+						create_ts: ctx.create_ts(),
+					})
+					.await?;
+				}
+			}
+
+			let last_event_idx = events.last().map(|event| event.index);
+
+			// NOTE: This should not be parallelized because signals should be sent in order
```

Comment:
add comment this is for deprecated actor events, in new version all actor events go through actor wf